typescript-rewrite/samlang-core/src/interpreter/__tests__/expression-interpreter.test.ts

```diff
@@ -0,0 +1,796 @@
+import {
```

Comment:
Tests look good, except that you need to add a test that related to the object constructor.
The tests would be more readable if you can utilize the existing parser and type checker infra. e.g.

```ts
const test = (sourceCode: string, expectedResult: Value) => {
   expect(interpret(typeCheck(parse(sourceCode)))).toEqual(expectedResult);
}
```

Although not push blocking for this diff.

Reply:
I think I do have object constructor tests on line 172?

Reply:
I think I have object constructor tests on line 172? Wow ya I was wondering if there was a better way to write these tests 😅

Reply:
ok then that looks good